### PR TITLE
feat: add layout component

### LINKED ISSUE #549: iPhoneX layout
Some minor tweaks need to be done.

See https://github.com/cozy/cozy-drive/pull/992/files

### PR DESCRIPTION
**Pull request by GoOz**
Added a `<Layout />` component to ease app integration and enforce consistency between different apps.

Preview: https://gooz.github.io/cozy-ui/react/#layout-1

Thorough reviews are very welcome because I'm not 100% sure I dealt with every use case.

Fixes #549

### CODE REVIEW

**Comment by GoOz** [comment]
[Ajouter un composant <Layout />](https://trello.com/c/PxhFODC5/19-ajouter-un-composant-layout)

File: stylus/objects/layouts.styl [hunk 1]
```diff
@@ -91,7 +92,8 @@ $app
         display block
 
         main,
-        main > [role=contentinfo]
+        main > [role=contentinfo], // Deprecated
```
Comment (ptbrowne) [inline stylus/objects/layouts.styl]: ping @y-lohse @goldoraf seems cozy-photos is using this.

**Comment by ptbrowne** [comment]
Does it support iPhoneX ?

File: react/Layout/Layout.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styles from './Layout.styl'
+import cx from 'classnames'
+
+export const Layout = ({ children, className, monoColumn, ...rest }) => {
+  return (
+    <div
+      className={cx(
+        monoColumn ? styles['o-layout'] : styles['o-layout--2panes'],
```
Comment (ptbrowne) [inline react/Layout/Layout.jsx]: I think o-layout--2panes should not be with `--` as it is not a variant. I would think a variant should be applied additionally to the normal class. In this case, it is one or the other.

File: react/Layout/Layout.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        </NavItem>
+      </Nav>
+    </Sidebar>
+    <Main>
+      <Content className='u-p-1'>
+        { content.ada.short }
+      </Content>
+    </Main>
+</Layout>
+```
```
Comment (ptbrowne) [inline react/Layout/Layout.md]: It is not possible to have it as a real example ?

File: stylus/objects/layouts.styl (hunk 1 above)
Comment (GoOz) [reply stylus/objects/layouts.styl]: Actually almost every app uses this, that's why I'm just commenting the deprecated status and not removing it completely yet.

**Comment by GoOz** [comment]
It does not. That wasn't the purpose of this PR but I guess I could add this too.

File: react/Layout/Layout.jsx (hunk 2 above)
Comment (GoOz) [reply react/Layout/Layout.jsx]: true. It used to be a variant but I didn't change the naming. 👍

File: react/Layout/Layout.md (hunk 3 above)
Comment (GoOz) [reply react/Layout/Layout.md]: I don't get it, what do you mean? Like real content like with one of our apps?

File: stylus/objects/layouts.styl (hunk 1 above)
Comment (y-lohse) [reply stylus/objects/layouts.styl]: Is it enough to just remove the `[role=contentinfo]` ? If so that would be a quick change to do everywhere.

**Comment by y-lohse** [comment]
I also think this would be a good place to integrate some iphone x changes :)

File: stylus/objects/layouts.styl (hunk 1 above)
Comment (GoOz) [reply stylus/objects/layouts.styl]: It's more replacing than removing but basically yeah, that's it.

File: react/Layout/Layout.md (hunk 3 above)
Comment (ptbrowne) [reply react/Layout/Layout.md]: sorry i was confused with the JSX code, I thought the example was not rendered, it's ok.

**Comment by GoOz** [comment]
@y-lohse Does it look good now?

**Comment by GoOz** [comment]
@y-lohse Time will tell… ;)

**Pull request merged**